test/source/test.ts

```diff
@@ -82,52 +75,67 @@ const testWithBrowser = (
         }
         await cb(t, browser);
         if (DEBUG_BROWSER_LOG) {
-          try {
-            const page = await browser.newPage(t, TestUrls.extension('chrome/dev/ci_unit_test.htm'));
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return
-            const items = (await page.target.evaluate(() => (window as any).Debug.readDatabase())) as {
-              input: unknown;
-              output: unknown;
-            }[];
-            for (let i = 0; i < items.length; i++) {
-              const item = items[i];
-              const input = JSON.stringify(item.input);
-              const output = JSON.stringify(item.output, undefined, 2);
-              const file = `./test/tmp/${t.title}-${i}.txt`;
-              writeFileSync(file, `in: ${input}\n\nout: ${output}`);
-              t.log(`browser debug written to file: ${file}`);
-            }
-          } catch (e) {
-            t.log(`Error reading debug messages: ${e}`);
-          }
+          await saveBrowserLog(t, browser);
         }
         t.log(`run time: ${Math.ceil((Date.now() - start) / 1000)}s`);
       },
       t,
       consts,
       flag
     );
+    if (t.closeMockApi) {
+      await t.closeMockApi();
+    }
```

Comment:
There should probably be a `try/finally` statement here, wrapping `withNewBrowserTimeoutAndRetry`. Then the `closeMockApi` would happen in `finally`. Then you can ensure it will run?

Reply:
Yeah, you're right, I added `try/finally` here